changelog.d/13801.feature

```diff
@@ -0,0 +1 @@
+Add `listeners[x].request_id_header` config to specify which request header to extract and use as the request ID in order to correlate timeouts in reverse-proxy layers in front of Synapse when tracing.
```

Comment:
I think this can ignore the timelines but and just say "in order to correlate requests from a reverse-proxy"?